src/Aspire.Hosting/ApplicationModel/UserSecretsParameterDefault.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Reflection;
+using Aspire.Hosting.Publishing;
+using Microsoft.Extensions.Configuration.UserSecrets;
+using Microsoft.Extensions.SecretManager.Tools.Internal;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Wraps a <see cref="ParameterDefault"/> such that the default value is saved to the project's user secrets store.
+/// </summary>
+/// <param name="appHostAssembly">The app host assembly.</param>
+/// <param name="applicationName">The application name.</param>
+/// <param name="parameterName">The parameter name.</param>
+/// <param name="parameterDefault">The parameter with a default value.</param>
+public sealed class UserSecretsParameterDefault(Assembly? appHostAssembly, string applicationName, string parameterName, ParameterDefault parameterDefault)
```

Comment:
Since this is public API, do we want to do any argument checking that things like `parameterDefault` are not null?

Reply:
We're super inconsistent with this in the repo right now and last time I brought it up the decision was not to bother adding it 'cos we have nullable annotations. I'm happy to add it but we really should decide and update it everywhere 😄

Reply:
Now I've moved the caller of this down to `ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter` we could choose to make it internal. Thoughts?

Reply:
> not to bother adding it 'cos we have nullable annotations

At least in dotnet/runtime, our stance is basically you should never allow your library to throw a NullReferenceException. A big reason is because it is harder to figure out how did the `null` get there. Whereas an ArgNullException catches it earlier.

It's definitely a drawback to primary constructors that you can't check for null, and I can see why the push for `!!` was so great.

> Now I've moved the caller of this down to ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter we could choose to make it internal. Thoughts?

Yeah, let's make it internal for now. We can always expose it later. (can't do the opposite)

Reply:
OK I made it internal